### PR TITLE
You've Got Mail

### PR DESCRIPTION
**Pull request by brianteeman**
image

Since 2003 the internet has changed. We no longer get a message to say that we have a message. Instead we just give you the message. You probably never use the messages component (especially for private message to a specific user) as they are the equivalent of https://www.youtube.com/watch?v=gFBLiHpkcOk

The Joomla com_messages component is used in two instances

1. Notification of a new article
2. Sending a message to another user

### Current email for Notification of a new article
**Subject:** A new private message has arrived from [sitename]
**Body:**
> Please log in to [link] to read your message.
***
### New email for Notification of a new article
**Subject:** New message from [user] at [sitename]
**Body:**
> New Article
A new Article has been submitted by 'user' entitled 'blog post'.
> Please log in to [link] to read your message.
***
### Current email when sending a message to another user
**Subject:** A new private message has arrived from [sitename]
**Body:**
> Please log in to [link] to read your message.
***
### New email when sending a message to another user
Subject: New message from [user] at [sitename]
**Body:**
> [subject]
 [message]
[login link]
***
## Backwards Compatibility
No issues. The message contains the old login message PLUS the content of the message. So if you were using this message in a custom workflow there is no change required to that workflow

### CODE REVIEW

File: administrator/components/com_messages/models/message.php [hunk 1]
```diff
@@ -359,9 +359,13 @@ public function save($data)
...
 			// Build the email subject and message
 			$sitename = JFactory::getApplication()->get('sitename');
+			$fromName = $fromUser->get('name');
 			$siteURL  = JUri::root() . 'administrator/index.php?option=com_messages&view=message&message_id=' . $table->message_id;
-			$subject  = sprintf($lang->_('COM_MESSAGES_NEW_MESSAGE_ARRIVED'), $sitename);
-			$msg      = sprintf($lang->_('COM_MESSAGES_PLEASE_LOGIN'), $siteURL);
+			$subject  = html_entity_decode($table->subject, ENT_COMPAT, 'UTF-8');
+			$message  = strip_tags(html_entity_decode($table->message, ENT_COMPAT, 'UTF-8'));
+
+			$subj	  = sprintf($lang->_('COM_MESSAGES_NEW_MESSAGE_ARRIVED'), $fromName, $sitename );
```
Comment (Quy) [inline administrator/components/com_messages/models/message.php]: Remove space ` );`

**Comment by infograf768** [comment]
I am not sure this is b/c because of the order of the variables in the sprintf.
is that so urgent it can’t wait 4.0?

**Comment by brianteeman** [comment]
So I can simply resolve that by making ti a completely new string and then there is no issue - will do that right now

**Comment by brianteeman** [comment]
PR updated

**Comment by ReLater** [comment]
I have tested this item :white_check_mark: successfully on 0e97189cb58c5be562027cbfc79a405ebc3ab69c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20162">issues.joomla.org/tracker/joomla-cms/20162</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 0e97189cb58c5be562027cbfc79a405ebc3ab69c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20162">issues.joomla.org/tracker/joomla-cms/20162</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20162">issues.joomla.org/tracker/joomla-cms/20162</a>.</sub>

**Comment by brianteeman** [comment]
thanks

**Pull request merged**